src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -359,7 +359,15 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 			if (strcmp(opt->defname, "admin") == 0)
 			{
 				appendStringInfo(buf, "ADMIN OPTION FOR ");
-				break;
+			}
+			else if (strcmp(opt->defname, "inherit") == 0);
```

Comment:
So, what I meant here https://github.com/citusdata/citus/pull/7156/files/3884dbe5e9adc162a886b3e8d4c471c74bd1e523..62762a7f209475cc0f7fd23923a3870fcefc6c23#r1308425694
is that in this loop we need to keep track of whether we have more than one option to be added, because the syntax doesn't allow for multiple WITH's.